File: packages/electron/src/main/net/messages/getStructType.ts
import { RROxApp } from "../../app";
import { BufferIO } from "../io";
import { MessageType, Request } from "../message";
import { GetStructResponse } from "./getStruct";

export class GetStructTypeRequest extends Request {
    public name: string;

    constructor( app: RROxApp, name: string ) {
        super( app, MessageType.GET_STRUCT_TYPE );

        this.name = name;
    }

    public process( data: BufferIO ) {
        super.process( data );

        data.writeString( this.name );
    }
}

export class GetStructTypeResponse extends GetStructResponse {}

File: packages/dll/src/net/messages/getstructtype.cpp
#include "getstructtype.h"
#include "../message.h"
#include "../../injector.h"
#include "../../UE/v425/uobjectarray.h"

GetStructTypeRequest::GetStructTypeRequest(Buffer& data) : Request(data), name(data.Read()) {}

void GetStructTypeRequest::Process() {
	GetStructTypeResponse res;

	injector.log("Retrieving struct type for: " + name);

	WFUObjectItem item = injector.objectArray.FindObject(name);

	if (item) {
		WUObject wrappedObj = item.GetObject();
		WUClass wrappedClass = wrappedObj.GetClass();
		res = GetStructTypeResponse(new GeneratedStruct(wrappedClass));
	}

	ProcessResponse(res);
}